addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -303,26 +304,43 @@ func (r *MirrorPeerReconciler) toggleCSIAddons(ctx context.Context, namespace st
 
 func (r *MirrorPeerReconciler) fetchClusterFSIDs(ctx context.Context, mp *multiclusterv1alpha1.MirrorPeer, clusterFSIDs map[string]string) error {
 	for _, pr := range mp.Spec.Items {
-		var secretName string
-		if pr.ClusterName == r.SpokeClusterName {
-			secretName = fmt.Sprintf("cluster-peer-token-%s-cephcluster", pr.StorageClusterRef.Name)
-		} else {
-			secretName = utils.GetSecretNameByPeerRef(pr)
-		}
-		klog.Info("Checking secret ", secretName)
-		secret, err := utils.FetchSecretWithName(ctx, r.SpokeClient, types.NamespacedName{Name: secretName, Namespace: pr.StorageClusterRef.Namespace})
+		clusterType, err := utils.GetClusterType(pr.StorageClusterRef.Name, pr.StorageClusterRef.Namespace, r.SpokeClient)
 		if err != nil {
-			klog.Error(err, "Error while fetching peer secret", "peerSecret ", secretName)
 			return err
 		}
+		var secretName string
+		if clusterType == utils.CONVERGED {
```

Comment:
CONVERGED and EXTERNAL have multiple if conditions to find the secret name and fetch ceph fsid. It is better to use different utils functions for these two.  It is easy to do changes when a new type is coming.

Reply:
Added utility functions for getting the secret and the fsid based on cluster type